### PR TITLE
Messaging and typed values (Unity side)

### PR DESCRIPTION
**Pull request by bosr**
Pursuing the idea in https://github.com/OpenMined/OpenMined/issues/166, I suggest that we upgrade the string messages returned by Unity to JSON messages like for instance

    {'ok': True, 'objectType': 'tensorID', 'id': 34}

Those messages are returned from the `processMessage` function in `FloatTensor.cs` using helpers like

- `asOK()` for signalling the success of an inline operation
- `asBool(false|true)` for returning the value of a property (think of `autograd`, `keepgrad`, etc)
- `asFloat(mytracevalue)` for returning a float
- `asID(result.Id)` for returning a TensorId
- `asIntList(...)` for shapes, etc

Those helper functions are currently defined in `FloatTensor.cs`, they create a `ResponseMessage` instance that is currently defined in the `SyftController.cs` file (actually there is a small hierarchy of helper classes to achieve the variety of messages above).

Basically, in the case that handles `round`:

```
case "round":
{
    var result = Round();
    // return result.Id.ToString();   // was a string
    return asID(result.id);              // <- becomes typed message
}
```

the case that handles `round_`:

```
case "round_":
{
    Round(inline: true);
    // return Id.ToString();  // returned a string
    return asOK();              // <- now returns a "OK" status message
}
```

and finally the case of the `autograd` property now is

```
case "autograd":
{
    // if (this.autograd)
    //     return "1";
    // return "0";
    return asBool(this.autograd);  // <- I'm sure you had guessed it
}

```

The corresponding JSON message for those messages are respectively:

```
{"ok":true,"objectType":"tensorID","id":34}
{"ok":true,"objectType":"status"}
{"ok":true,"objectType":"bool","value":true}
```

In case of failure, an error message looks like
```
{"ok":false,"error":"the error message"}
```

I submit a corresponding PR in Syft where that message handling is implemented, and some semantics are added to operations (`_op`, `_op_inline`, `_get_prop`, `_manage`, ...).

Because it only affects messaging, no tests are broken here. Only integration tests may be affected. But rest that almost (I think all) calls in PySyft have been updated for this.

I'm really eager to hear your comments.

PS: I intentionally left the commented previous return values, only temporarily for the update to be clearer if you directly read the code. I intend to remove those after having received your remarks.

### CODE REVIEW

**Comment by iamtrask** [comment]
Working on the merge now!

**Comment by bosr** [comment]
tldr; there are no API changes, all methods remain the same, only messaging with Syft have changes

**Comment by bosr** [comment]
rebased: but I now need to extend the messages to the new `Model`, it won't compile for now.

Could you please add a work-in-progress flag?

**Comment by bosr** [comment]
I have the strong feeling that this PR does not come at the right time (details available on demand).

No problem, I'll find some other issue to work on!
